eox_tenant/auth.py

```diff
@@ -0,0 +1,82 @@
+
+"""
+This file implements the authentication backend for the openedx platform.
+"""
+import logging
+import re
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+
+from django.utils.translation import ugettext as _
+
+from eox_tenant.edxapp_wrapper.auth import get_edx_auth_backend, get_edx_auth_failed
+
+AuthFailedError = get_edx_auth_failed()
+AUDIT_LOG = logging.getLogger("audit")
+EdxAuthBackend = get_edx_auth_backend()
+UserModel = get_user_model()
+
+
+class TenantAwareAuthBackend(EdxAuthBackend):
+    """
+    Authentication Backend class which will check if the user has a signupsource in the requested site.
+    """
+    def authenticate(self, request, username=None, password=None, **kwargs):
+        """
+        Override default method to store the request which is being used by the
+        user_can_authenticate_on_tenant method.
+        """
+        self.request = request  # pylint: disable=attribute-defined-outside-init
```

Comment:
just leave a comment here indicating we are doing this to use the request in a method the variable is not accessible

Reply:
I added that to the docstring of the method, should I add that here too?